components/utils/strictmodeviolator/src/main/java/org/mozilla/strictmodeviolator/StrictModeViolation.kt

```diff
@@ -0,0 +1,18 @@
+package org.mozilla.strictmodeviolator
+
+import android.os.StrictMode
+
+class StrictModeViolation {
+
+    companion object {
+        fun <T> tempGrant(tempChange: (StrictMode.ThreadPolicy.Builder) -> StrictMode.ThreadPolicy.Builder, lambda: () -> T): T {
+            val cachedPolicy = StrictMode.getThreadPolicy()
+            StrictMode.setThreadPolicy(tempChange.invoke(StrictMode.ThreadPolicy.Builder(cachedPolicy)).build())
+            try {
```

Comment:
this `try` is interesting.... is it really needed? seems to be yes...

Reply:
I think it's a lot safer. please not that It's not catched though.